apps/meteor/client/views/room/contextualBar/Info/RoomInfo/RoomInfoWithData.js

```diff
@@ -68,9 +72,13 @@ const RoomInfoWithData = ({ rid, openEditing, onClickBack, onEnterRoom, resetSta
 		t('Success'),
 	);
 
-	const canDelete = usePermission(type === 'c' ? 'delete-c' : 'delete-p', rid);
-	const canEdit = usePermission('edit-room', rid);
-	const canConvertRoomToTeam = usePermission('create-team');
+	const isFederated = isRoomFederated(room);
+	const hasPermissionToDelete = usePermission(type === 'c' ? 'delete-c' : 'delete-p', rid);
+	const hasPermissionToEdit = usePermission('edit-room', rid);
+	const hasPermissionToConvertRoomToTeam = usePermission('create-team');
+	const canDelete = isFederated ? Federation.canEdit(user, room) : hasPermissionToDelete;
+	const canEdit = isFederated ? Federation.canEdit(user, room) : hasPermissionToEdit;
```

Comment:
I think even for federated rooms we should check the permission, wdyt?

Reply:
Very good question, but yes, you are right. Probably in the near future when we'll start to have some sort of overlap between RC roles and Matrix roles, we need to revisit that, but for now, I agree 👍